### PR TITLE
[IMP] mail: fetch more messages on thread

### PR DESCRIPTION
**Pull request by alexkuhn**
add a button link to load more messages on a thread

### CODE REVIEW

**Comment by alexkuhn** [comment]
No test for the moment, follow-up PR will put auto-load on scroll back, which will allow to re-enable some old tests

File: addons/mail/static/src/new/messaging.js [hunk 1]
```diff
@@ -307,10 +310,15 @@ export class Messaging {
         } else {
             thread.messages.push(id);
         }
+        this.sortThreadMessages(thread);
```
Comment (ged-odoo) [inline addons/mail/static/src/new/messaging.js:313]: it feels like this is wrong for chatter messages: the code above sort then in reverse order.

Also, note that this is a quadratic behaviour: adding n messages means that we have to sort the messages list n times. probably not a big deal in practice, but usually we can do better

Comment (ged-odoo) [reply addons/mail/static/src/new/messaging.js:313]: ok, it seems like it works thanks to the order props. not a fan of having to reverse it each time, but it works

**Review by ged-odoo — APPROVED** [review]
really cool

File: addons/mail/static/src/new/messaging.js (hunk 1 above)
Comment (alexkuhn) [reply addons/mail/static/src/new/messaging.js:313]: Agree, we could find a way to save both orders.
Note that chatter can be in a chat window, and the order should be the same as with channels. Hence why we can't just keep a single ordering of messages for a given thread.

**Pull request merged**